Modules/posixmodule.c

```diff
@@ -6307,11 +6307,7 @@ os_utime_impl(PyObject *module, path_t *path, PyObject *times, PyObject *ns,
         _Py_time_t_to_FILE_TIME(utime.mtime_s, utime.mtime_ns, &mtime);
     }
     if (!SetFileTime(hFile, NULL, &atime, &mtime)) {
-        /* Avoid putting the file name into the error here,
-           as that may confuse the user into believing that
-           something is wrong with the file, when it also
-           could be the time stamp that gives a problem. */
-        PyErr_SetFromWindowsErr(0);
+        PyErr_SetFromWindowsErrWithFilename(0, path->wide);
```

Comment:
You don't pass the right filename type to PyErr_SetFromWindowsErrWithFilename(). Try to use posix_path_object_error() instead.

Reply:
Err, woops, yeah this shouldn't be wide. I have no easy way to test this on windows.

I'll change the posix section to posix_path_object_error().

Reply:
I don't understand. You didn't fix the bug in the Windows code path. I suggest using posix_path_object_error() in the Windows code path (as well?).

Reply:
Erm, will that work? I'm not too familiar with the python C api or Windows platform tbh, but from [1] it seems like the *Windows functions use some Windows api (GetLastError) and the posix ones will refer to errno.

Looking again, it seems like posix_path_error() and posix_path_object_error() are identical in this case, so I'm definitely confused.

... and as I was typing this I think I understand: do you mean to use `path_error/path_object_error`? It seems like they will do the handling correctly for both platforms.

[1] https://docs.python.org/3/c-api/exceptions.html#c.PyErr_SetFromWindowsErr

Reply:
I don't have easy access to Windows or a Windows C compiler but this method still works as intended on Linux at least.